codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -202,7 +200,7 @@ public void operationComplete(ChannelFuture future) throws Exception {
                 // Pass headers to the flow-controller so it can maintain their sequence relative to DATA frames.
                 flowController.addFlowControlled(stream,
                         new FlowControlledHeaders(stream, headers, streamDependency, weight, exclusive, padding,
-                                                  endOfStream, promise));
+                                                  true, promise));
```

Comment:
```java
if (!endOfStream || !flowController.hasFlowControlled(stream)) {
} else {
  // endOfStream cannot be false here (we would have fallen into the if block above).
}
```

Am I missing something? As I said I can revert if you prefer as it is not necessary for this PR.